contracts/reserve/GoodReserveCDai.sol

```diff
@@ -266,6 +368,73 @@ contract GoodReserveCDai is
 		);
 		return gdReturn;
 	}
+	/**
+	 * @dev Converts GD tokens to `sellTo` tokens and update the bonding curve params.
+	 * `sell` occurs only if the token return is above the given minimum. Notice that
+	 * there is a contribution amount from the given GD that remains in the reserve.
+	 * It is only possible to sell to cDAI and only when the contract is set to
+	 * active. MUST be called to G$ `approve` prior to this action to allow this
+	 * contract to accomplish the conversion.
+	 * @param _sellTo The tokens that will be received after the conversion
+	 * @param _gdAmount The amount of GD tokens that should be converted to `_sellTo` tokens
+	 * @param _minReturn The minimum allowed `sellTo` tokens return
+	 * @param _minTokenAmount The mininmum dai out amount from Exchange swap function
+	 * @param _targetAddress address of _sellTo token recipient if different than msg.sender
+	 * @return (tokenReturn) How much `sellTo` tokens were transferred
+	 */
+	function sellToAnyToken(
+		ERC20 _sellTo,
+		uint256 _gdAmount,
+		uint256 _minReturn,
+		uint256 _minTokenAmount,
+		address _targetAddress
+	) public returns (uint256) {
+		
+		address cDaiAddress = nameService.getAddress("CDAI");
+		address daiAddress = nameService.getAddress("DAI");
+		// redeclarations to prevent stack too deep errors
+		ERC20 tempSellTo = _sellTo;
+		uint256 tempMinReturn = _minReturn;
+		uint256 tempGdAmount = _gdAmount;
+		if(address(_sellTo) == cDaiAddress){
+			(uint256 result,uint256 contributionAmount) = sell(_sellTo, _gdAmount, _minReturn, true, false, _targetAddress);
```

Comment:
why withTransfer is false?

Reply:
When you call sell with withTransfer is false , it doesn't transfer DAI so we can use it for further exchanges

Reply:
yes but in the next line you "return" so you dont use it anywhere for further exchanges
how is this passing unit tests?